Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -667,16 +668,7 @@ - (void)willChange:(NSKeyValueChange)changeKind valuesAtIndexes:(NSIndexSet*)ind
 
             if ((options & NSKeyValueObservingOptionOld) && localChangeKind != NSKeyValueChangeInsertion) {
                 // For to-many mutations, we can't get the old values at indexes that have not yet been inserted.
-                change[NSKeyValueChangeOldKey] =
-                    _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keypath, keyObserver, indexes);
-            }
-
-            if (options & NSKeyValueObservingOptionPrior) {
-                id observer = keypathObserver.observer;
-                void* context = keypathObserver.context;
-                [change setObject:@(YES) forKey:NSKeyValueChangeNotificationIsPriorKey];
-                [observer observeValueForKeyPath:keypath ofObject:rootObject change:change context:context];
-                [change removeObjectForKey:NSKeyValueChangeNotificationIsPriorKey];
+                change[NSKeyValueChangeOldKey] = _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keyObserver, indexes);
```

Comment:
oh, i understand what was happening. do you know if this holds up with the comment on 662? We're degrading a collection mutation event to a single setter mutation event. `valueForPendingChange...` was supposed to _explicitly not_ do `objectsAtIndexes:` in that case.

It was intended to fall back to the `!indexes` case (original code, line 524)